jax/_src/cudnn/fused_attention_stablehlo.py

```diff
@@ -28,14 +28,26 @@
 from jax.experimental.custom_partitioning import custom_partitioning
 from jax.sharding import PartitionSpec, NamedSharding
 
+from jax._src import api
+from jax._src import config
 from jax._src import dispatch
+from jax._src import lib
+from jax._src import util
 from jax._src.interpreters import batching
 from jax._src.lib import cuda_versions
+from jax._src.lib import xla_extension_version
 
 Array = jnp.ndarray
 DType = jnp.dtype
 PRNGKey = jnp.ndarray
 
+def _normalize_layout(layout_str):
```

Comment:
I think choosing between btnh and bnth is good enough.

Reply:
Passing the str layout makes the API more generic and scalable. But, ss you suggested, I make the internal representation as an enum of two possible values. This can also help us avoid passing str into the jit function, which can help us avoid the hassle of the intricacy of setting `static_argnum/names`.